Objective-C/Internal/Replicator/CBLTrustCheck.mm

```diff
@@ -186,7 +186,7 @@ - (NSURLCredential*) checkTrust: (NSError**)outError {
             NSData* certData = nil;
             for (CFIndex i = 0; i < count; i++) {
                 CFArrayRef certs = SecTrustCopyCertificateChain(_trust);
-                SecCertificateRef cert = (SecCertificateRef)CFArrayGetValueAtIndex(certs, 0);
+                SecCertificateRef cert = (SecCertificateRef)CFArrayGetValueAtIndex(certs, i);
```

Comment:
Does this mean that the pinned cert is still matched with the leaf cert? 

I think recently there is a question from support about this as a customer has been asked for the feature, and I have informed the support that we do support this. We may need to port the fix to 3.0.x and raise this to Priya and support?

Reply:
Related CBSE-12463 (Show the customer and support who were informed about the feature)

Reply:
I assume, this won't happen to any released product, since this flow will work only with Xcode 13+, before it was going through the below check. That's why the test was passing before updating the Xcode in build machine. 


https://github.com/couchbase/couchbase-lite-ios/blob/9e0451c2d847b79b03930544877d69f9111af318/Objective-C/Internal/Replicator/CBLTrustCheck.mm#L201